packages/studiocms_devapps/src/utils/wp-api/converters.ts

```diff
@@ -76,17 +78,114 @@ export const ConvertToPageContent = async (
 	return pageContent;
 };
 
-export const ConvertToPostData = async (post: unknown, useBlogPkg: boolean): Promise<PageData> => {
+export const generateCatagories = async (categories: number[], endpoint: string) => {
+	const newCatagories: Category[] = [];
+
+	for (const catagoryId of categories) {
+		// Check if catagory already exists in the database
+		const catagoryExists = await db
+			.select()
+			.from(tsPageDataCategories)
+			.where(eq(tsPageDataCategories.id, catagoryId))
+			.get();
+
+		if (!catagoryExists) {
+			const catagoryURL = apiEndpoint(endpoint, 'catagories', `${catagoryId}`);
+			const response = await fetch(catagoryURL);
+			const json = await response.json();
+			newCatagories.push(json);
+		}
+
+		if (catagoryExists) {
+			console.log(`Catagory with id ${catagoryId} already exists in the database`);
+		}
```

Comment:
I always prefer to exit early when checking existence

```ts
const categoryExists = ...

if (categoryExists) {
  console.log('sorry fam')
  continue
}

// do actual stuff
```

Reply:
[6a00b06](https://github.com/astrolicious/studiocms/pull/367/commits/6a00b066dc936f429d2a3b70978a308ff0b2e4d4) done